lotus/sem_ops/sem_join.py

```diff
@@ -231,7 +524,13 @@ def __call__(
         examples: pd.DataFrame | None = None,
         strategy: str | None = None,
         default: bool = True,
-        cascade_threshold: float | None = None,
+        recall_target: float | None = None,
```

Comment:
If all these parameters are meant to be in a group it might be more clear to have something like a `SemJoinCascadeArgs` model which contains all this information. This way we know exactly what is needed for cascade.